portal-kernel/src/com/liferay/portal/kernel/upgrade/BaseBatchImportStatusUpgradeProcess.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.kernel.upgrade;
+
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.util.LoggingTimer;
+import com.liferay.portal.kernel.util.StringBundler;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+/**
+ * @author Alessio Antonio Rendina
+ */
+public abstract class BaseBatchImportStatusUpgradeProcess
+	extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		for (String[] tableAndPrimaryKeyColumnName :
+				getTableAndPrimaryKeyColumnNames()) {
+
+			String tableName = tableAndPrimaryKeyColumnName[0];
+			String primKeyColumnName = tableAndPrimaryKeyColumnName[1];
+
+			upgradeBatchImportStatus(tableName, primKeyColumnName);
+		}
+	}
+
+	protected abstract String[][] getTableAndPrimaryKeyColumnNames();
+
+	protected void upgradeBatchImportStatus(
+			String tableName, String primKeyColumnName)
+		throws Exception {
+
+		if (!hasTable(tableName)) {
+			_log.error("Skip nonexistent table " + tableName);
+
+			return;
+		}
+
+		if (_log.isInfoEnabled()) {
+			_log.info("Upgrade table " + tableName);
+		}
+
+		if (!hasColumn(tableName, "batchImportStatus")) {
+			alterTableAddColumn(tableName, "batchImportStatus", "INTEGER");
+		}
+
+		try (LoggingTimer loggingTimer = new LoggingTimer()) {
+			StringBundler selectSB = new StringBundler(4);
+
+			selectSB.append("select ");
+			selectSB.append(primKeyColumnName);
+			selectSB.append(" from ");
+			selectSB.append(tableName);
+
+			StringBundler updateSB = new StringBundler(5);
+
+			updateSB.append("update ");
+			updateSB.append(tableName);
+			updateSB.append(" set batchImportStatus = 0 where ");
+			updateSB.append(primKeyColumnName);
+			updateSB.append(" = ?");
+
+			try (PreparedStatement preparedStatement1 =
+					connection.prepareStatement(selectSB.toString());
+				ResultSet resultSet = preparedStatement1.executeQuery();
+				PreparedStatement preparedStatement2 =
+					AutoBatchPreparedStatementUtil.autoBatch(
+						connection, updateSB.toString())) {
+
+				while (resultSet.next()) {
+					long primKey = resultSet.getLong(1);
+
+					preparedStatement2.setLong(1, primKey);
+
+					preparedStatement2.addBatch();
+				}
+
+				preparedStatement2.executeBatch();
+			}
```

Comment:
I think this can be simplify because the query needed should be just
```
update $tableName set batchImportStatus = 0 where batchImportStatus is null
```

we don't need to loop on all the rows but when we run this upgrade process ALL the rows without a value (if I run the upgrade process multiple time) get the value 0